proto/cosmos/bank/query.proto

```diff
@@ -9,67 +9,69 @@ option go_package = "github.com/cosmos/cosmos-sdk/x/bank/types";
-    bytes address = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+  // address is the address to query balances for
+  bytes address = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
 
-    // denom is the coin denom to query balances for
-    string denom = 2;
+  // denom is the coin denom to query balances for
+  string denom = 2;
 }
 
 // QueryBalanceResponse is the response type for the Query/Balance RPC method
 message QueryBalanceResponse {
-
-    // balance is the balance of the coin
-    cosmos.Coin balance = 1;
+  // balance is the balance of the coin
+  cosmos.Coin balance = 1;
 }
 
 // QueryBalanceRequest is the request type for the Query/AllBalances RPC method
 message QueryAllBalancesRequest {
-    // address is the address to query balances for
-    bytes address = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+  // address is the address to query balances for
+  bytes address = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
 
-    cosmos.query.PageRequest req = 2;
+  cosmos.query.PageRequest req = 2;
 }
 
 // QueryAllBalancesResponse is the response type for the Query/AllBalances RPC method
 message QueryAllBalancesResponse {
-    // balances is the balances of the coins
-    repeated cosmos.Coin balances = 1 [(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
+  // balances is the balances of the coins
+  repeated cosmos.Coin balances = 1
+      [(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
 
-    cosmos.query.PageResponse res = 2;
+  cosmos.query.PageResponse res = 2;
 }
 
 // QueryTotalSupplyRequest is the request type for the Query/TotalSupply RPC method
 message QueryTotalSupplyRequest {}
 
 // QueryTotalSupplyResponse is the response type for the Query/TotalSupply RPC method
 message QueryTotalSupplyResponse {
-    // supply is the supply of the coins
-    repeated cosmos.Coin supply = 1 [(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
+  // supply is the supply of the coins
+  repeated cosmos.Coin supply = 1
+      [(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
 }
 
 // QuerySupplyOfRequest is the request type for the Query/SupplyOf RPC method
 message QuerySupplyOfRequest {
-    string denom = 1;
+  string denom = 1;
 }
 
 // QuerySupplyOfResponse is the response type for the Query/SupplyOf RPC method
 message QuerySupplyOfResponse {
-    // amount is the supply of the coin
-    string amount = 1 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int", (gogoproto.nullable) = false];
+  // amount is the supply of the coin
+  cosmos.Coin amount = 1
+      [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Coin", (gogoproto.nullable) = false];
```

Comment:
This returns a `sdk.Coin`, which matches the querier behavior and other queries similar to this.